### PR TITLE
Cache delegates - Closes #1103

### LINKED ISSUE #1103: Cache delegates
### Description

To allow users to filter the delegates, it is necessary to cache the delegate information.

### Acceptance Criteria

- All the delegates are fetched at init and cached in-memory
- The delegate cache is updated whenever the state is modified concerning a delegate

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1103 

### How was it solved?
- [x] Add job for delegates
- [x] Refactoring existing code for reloading delegates
- [x] Rename forgers to generators
- [x] The delegate cache is updated whenever the state is modified concerning a delegate

### How was it tested?
Local

### CODE REVIEW

File: services/blockchain-indexer/shared/dataService/business/delegates.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+const { requestConnector } = require('../../utils/request');
+
+const config = require('../../../config');
+
+const latestBlockCache = CacheRedis('latestBlock', config.endpoints.cache);
+
+const isDposModuleRegistered = async () => {
+	const networkStatus = await requestConnector('getNetworkStatus');
+	const isRegistered = networkStatus.registeredModules.some(module => module.id === 13);
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/business/delegates.js:32]: Move moduleID to config file

File: services/blockchain-indexer/shared/dataService/business/delegates.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	return isRegistered;
+};
+
+const verifyIfPunished = async delegate => {
+	const latestBlockString = await latestBlockCache.get('latestBlock');
+	const latestBlock = latestBlockString ? JSON.parse(latestBlockString) : {};
+	const isPunished = delegate.pomHeights
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/business/delegates.js:39]: Add TODO to get this info from SDK once available

File: services/blockchain-indexer/shared/dataService/delegates.js [hunk 3]
```diff
@@ -256,27 +259,27 @@ const getDelegates = async params => {
 // Keep the delegate cache up-to-date
 const updateDelegateListEveryBlock = () => {
 	const updateDelegateCacheListener = async (eventType, data) => {
-		const dposModuleId = 5;
-		const registerDelegateAssetId = 0;
-		const voteDelegateAssetId = 1;
+		const dposModuleID = 13;
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/delegates.js:262]: Read from config these values

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-indexer/shared/dataService/delegates.js [hunk 4]
```diff
@@ -81,14 +82,14 @@ const computeDelegateStatus = async () => {
 		logger.debug('Determine delegate status');
 
 		// Default delegate status
-		delegate.status = delegateStatus.NON_ELIGIBLE;
+		delegate.status = delegateStatus.INELIGIBLE;
 
 		// Update delegate status, if applicable
-		if (delegate.dpos.delegate.isBanned) {
+		if (delegate.isBanned) {
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/blockchain-indexer/shared/dataService/delegates.js:88]: Yes, i verified and `isBanned` is directly available in the response.

**Review by sameersubudhi — COMMENTED** [review]
This PR is approved!

**Pull request merged**